android/build.gradle.nix

```diff
@@ -72,6 +73,37 @@ android {
 
     packagingOptions {
     }
+
+    // see https://developer.android.com/studio/build/configure-apk-splits.html
+    // for information about this and the applicationVariants stuff below.
+    // See https://developer.android.com/google/play/publishing/multiple-apks.html#SingleAPK
+    // for reasons you might not want to do this.
+    ${if universalApk then "" else ''
+        splits {
+            abi {
+                enable true
+                reset()
+                include "armeabi-v7a", "arm64-v8a"
+                universalApk false
+            }
+        }
+        ''
+    }
+}
+
+ext.abiCodes = ['armeabi-v7a': 1, 'arm64-v8a': 2] // This order is important!
+
+import com.android.build.OutputFile
+
+android.applicationVariants.all { variant ->
+  variant.outputs.each { output ->
+    def baseAbiVersionCode =
+      project.ext.abiCodes.get(output.getFilter(OutputFile.ABI))
+
+    if (baseAbiVersionCode != null) { // this will be null if splitting was disabled
+      output.versionCodeOverride = baseAbiVersionCode * 1000 + variant.versionCode
```

Comment:
Does this encoding have any ramifications for someone who has published to the Play store switching to or away from universal APKs?  E.g., is going backwards in version code to switch back to universal OK?

Reply:
Yes, it does.  Version codes have to be strictly increasing, so it's much easier to switch FROM universal to switching TO them.

Something I considered doing is changing that `1000` to `2` so that version numbers would be much more tightly packed.  That way, if you wanted to switch back to universal, you'd have to set your version number to `2*(old_vc+1)`.  That'd be fairly future-proof too; if reflex grew support for, say, MIPS devices, the number there would become `3` and there'd be a skip in the version counting but not a giant one.

Reply:
I just read up on this, and I see that it's the recommended way of doing it, so I'm happy to go with it!  Anyone who needs to switch back can just increment their version code a *lot* - e.g. by 10000.  Gross, but it'll do.  Thanks for explaining!